profiler/src/ProfilerEngine/Datadog.Profiler.Native.Linux/LinuxStackFramesCollector.cpp

```diff
@@ -90,6 +88,51 @@ void LinuxStackFramesCollector::UpdateErrorStats(std::int32_t errorCode)
     }
 }
 
+bool LinuxStackFramesCollector::IsProfilerSignalHandlerInstalled()
+{
+    struct sigaction currentAction;
+
+    sigaction(s_signalToSend, nullptr, &currentAction);
+
+    return currentAction.sa_sigaction == LinuxStackFramesCollector::CollectStackSampleSignalHandler;
+}
+
+std::int64_t LinuxStackFramesCollector::SendSignal(pid_t threadId) const
+{
+    return syscall(SYS_tgkill, _processId, threadId, s_signalToSend);
+}
+
+bool LinuxStackFramesCollector::CheckSignalHandler()
+{
+    if (IsProfilerSignalHandlerInstalled())
+    {
+        return true;
+    }
+
+    if (!_canReplaceSignalHandler)
+    {
+        static bool alreadyLogged = false;
+        if (alreadyLogged)
+            return false;
+
+        alreadyLogged = true;
+        Log::Warn("Profiler signal handler was replaced again. As of now, we will stopped restoring it to avoid issues.");
```

Comment:
Maybe more explicit like: ... to avoid issues: the profiler is disabled.

Reply:
✔️